torch/utils/_pytree.py

```diff
@@ -97,6 +97,7 @@ def _register_pytree_node(
     to_str_fn: Optional[ToStrFunc] = None,
     maybe_from_str_fn: Optional[MaybeFromStrFunc] = None,
     *,
+    type_fqn: Optional[str] = None,
```

Comment:
`type_fqn` seems a little opaque as name for this concept. A user may even be tempted to set this to the type's fqn, which would be wrong lol. 

Maybe: `serialized_type_name`?